lib/helpers/index.js

```diff
@@ -0,0 +1,59 @@
+exports = module.exports = {
+    /**
+     * Validate that response header exists
+     * @private
+     * @param {object} request
+     */
+    hasHeaders: function (request) {
+        let value = false;
+
+        if (typeof request.response === 'object' &&
+            request.response !== null &&
+            typeof request.response.header === 'function') {
+            value = true;
+        }
+
+        return value;
+    },
```

Comment:
if it were to come up false then custom headers wouldn't be set... in lib/config {setCustomHeaders}